File: frontend/src/types/playlist.ts
import type { Video } from "./video";

export enum PlaylistType {
    "Custom",
    "Liked",
    "WatchLater",
}

export interface Playlist {
    id: string;
    title: string;
    isPrivate: boolean;
    isSystem: boolean;
    type: PlaylistType;
    createdAt: string;
    lastUpdatedAt: string;
    userId: string;
    playlistVideos: { id: string; videoId: string }[];
}

export interface PlaylistVideoItem {
    id: string;
    videoId: string;
    videoTitle: string;
    channelName?: string;
    thumbnailUrl?: string;
    durationSeconds?: number;
    isDeleted?: boolean;
    addedAt: string;
    order: number;
    video?: Video;
}

export interface PlaylistResponse {
    id: string;
    title: string;
    isPrivate: boolean;
    isSystem: boolean;
    type: PlaylistType;
    videoCount: number;
    thumbnailUrl?: string;
    createdAt: string;
    lastUpdatedAt: string;
    videos: PlaylistVideoItem[];
}

File: backend/Youtube/Models/Playlist.cs
using System.ComponentModel.DataAnnotations;

namespace Youtube.Models
{
    public class Playlist
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        [MaxLength(150)]
        public string Title { get; set; } = string.Empty;

        public bool IsPrivate { get; set; } = false;
        public bool IsSystem { get; set; } = false; 
        public PlaylistType Type { get; set; } = PlaylistType.Custom;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime LastUpdatedAt { get; set; } = DateTime.UtcNow;

        public Guid UserId { get; set; }
        public User User { get; set; } = null!;

        public ICollection<PlaylistVideo> PlaylistVideos { get; set; } = new List<PlaylistVideo>();
    }

    public enum PlaylistType
    {
        Custom,       
        Liked,        
        WatchLater    
    }
}
